cdap-ui/app/cdap/components/PropertiesEditor/index.js

```diff
@@ -0,0 +1,259 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import {MyMetadataApi} from 'api/metadata';
+import NamespaceStore from 'services/NamespaceStore';
+import map from 'lodash/map';
+import shortid from 'shortid';
+import AddPropertyModal from 'components/PropertiesEditor/AddPropertyModal';
+import T from 'i18n-react';
+
+require('./PropertiesEditor.scss');
+
+const convertObjToArr = (obj) => {
+  let properties = map(obj, (value, key) => ({key, value}))
+    .map((row) => {
+      row.id = shortid.generate();
+      return row;
+    });
+  return properties;
+};
+
+export default class PropertiesEditor extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      systemProperties: [],
+      userProperties: [],
+      activeEdit: null,
+      newValue: ''
+    };
+
+    this.fetchUserProperties = this.fetchUserProperties.bind(this);
+    this.handleNewValueChange = this.handleNewValueChange.bind(this);
+  }
+
+  componentWillMount() {
+    let namespace = NamespaceStore.getState().selectedNamespace;
+    const baseRequestObject = {
+      namespace,
+      entityType: this.props.entityType,
+      entityId: this.props.entityId
+    };
+
+    let systemParams = Object.assign({}, baseRequestObject, { scope: 'SYSTEM' });
+    let userParams = Object.assign({}, baseRequestObject, { scope: 'USER' });
+
+    MyMetadataApi.getProperties(systemParams)
+      .map(convertObjToArr)
+      .combineLatest(MyMetadataApi.getProperties(userParams).map(convertObjToArr))
+      .subscribe((res) => {
+        this.setState({
+          systemProperties: res[0].filter((row) => row.key !== 'schema'),
+          userProperties: res[1]
+        });
+      });
+  }
+
+  componentDidUpdate() {
+    if (this.state.activeEdit && this.state.newValue.length === 0) {
+      document.getElementById(this.state.activeEdit).focus();
+    }
+  }
+
+  fetchUserProperties() {
+    let namespace = NamespaceStore.getState().selectedNamespace;
+    const params = {
+      namespace,
+      entityType: this.props.entityType,
+      entityId: this.props.entityId,
+      scope: 'USER'
+    };
+
+    MyMetadataApi.getProperties(params)
+      .map(convertObjToArr)
+      .subscribe((res) => {
+        this.setState({
+          userProperties: res,
+          activeEdit: null,
+          newValue: ''
+        });
+      });
+  }
+
+  deleteProperty(key) {
+    let namespace = NamespaceStore.getState().selectedNamespace;
+    const params = {
+      namespace,
+      key,
+      entityType: this.props.entityType,
+      entityId: this.props.entityId,
+    };
+
+    MyMetadataApi.deleteProperty(params)
+      .subscribe(this.fetchUserProperties, (err) => {
+        console.log('Error', err);
```

Comment:
I'm curious why you are logging the error here, and on line 131, but not in the `fetchUserProperties` function as well? Also, is it ok to just console log the error instead of displaying it?

Other than that, lgtm 👍